src/services/task1.ts

```diff
@@ -0,0 +1,33 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { File, FileFormBody } from "../types";
+
+export const task1Api = createApi({
+  reducerPath: "task1Api",
+  baseQuery: fetchBaseQuery({
+    baseUrl: "http://localhost:8080/api/v1/",
+  }),
+  endpoints: (builder) => ({
+    getFiles: builder.query<string[], void>({
+      query: () => "files",
+      transformResponse: ({ files }) => files,
+    }),
+    getFileByName: builder.query<File, string | undefined>({
```

Comment:
You shouldn't expect undefined here, in case there's no file you get 404 error

Reply:
Nvm, it's request param
Undefined as request param still shouldn't be a valid case